tools/hsmtool.c

```diff
@@ -754,14 +762,45 @@ int main(int argc, char *argv[])
 	}
 
 	if (streq(method, "dumponchaindescriptors")) {
+		char *fname = NULL;
 		char *net = NULL;
+		bool show_secrets = false;
+		bool only_arguments = false;
 		u32 version;
 
 		if (argc < 3)
 			show_usage(argv[0]);
 
-		if (argc > 3)
-			net = argv[3];
+		for (int i = 2; i < argc; ++i) {
+			char *next = argv[i];
+
+			if (only_arguments || next[0] != '-') {
+				// this is an argument
+				if (!fname) {
+					fname = next;
+					continue;
+				}
+				if (!net) {
+					net = next;
+					continue;
+				}
+				errx(ERROR_USAGE,
+				     "Argument '%s' was not expected.", next);
+			}
+
+			if (streq(next, "--")) {
+				only_arguments = true;
+				continue;
+			}
+
+			// we are processing an option here
+			if (streq(next, "--show-secrets")) {
+				show_secrets = true;
+				continue;
+			}
+			errx(ERROR_USAGE, "Option '%s' is not recognized.",
+			     next);
+		}
```

Comment:
I would love to. But IMO that deserves a separate PR.
BTW I was looking a bit at `ccan/opt`: it can be used to parse options, but it is not clear to me if it can handle positional arguments and commands.

Reply:
make sense